### ISSUE TITLE
Compile third party JS with Node.js source

### ISSUE THREAD

**Issue by saurabh-deep** [nodejs/help#3162]
* **Node.js Version**: 12.x / 14.x
* **OS**: Mac OS, Windows, Linux (Ubuntu)
* **Scope (install, code, runtime, meta, other?)**: runtime / packaging
* **Module (and version) (if relevant)**:

We are creating a Command Line Interface (CLI) using Node.js. We have written a lot of custom Node.js modules for this CLI including some for authentication. We want to compile Node.js from source and include all our custom code in the final binary so that we can ship a single executable and not worry about the users being able to see (and thus bypass the authentication).

In short, I am looking for a way to embed third party JS and compile the same with Node.js itself. We have tried [pkg](https://github.com/vercel/pkg) and [nexe](https://github.com/nexe/nexe), but the binaries produced by them are not valid for MacOS and therefore, cannot be signed.

Does Node.js 12/14 provide a way to achieve this? Much appreciate your help on this.

**Comment by saurabh-deep** [comment]
Hey All!

I am not trying to compare Node and Deno, but Deno now supports [this](https://github.com/denoland/deno/pull/8539) OOTB. May be there is a hidden way (or something apparent that I am not aware about) to do the same / similar in Node.js as well.

Eagerly looking forward to a response / suggestion.

Cheers!
Saurabh

**Comment by NickNaso** [comment]
Hi @saurabh-deep,
you can achieve at your purpose using some tools like:
- [boxnode](https://github.com/mongodb-js/boxednode)
- [pkg](https://github.com/vercel/pkg)

I hope that this help you.

**Comment by saurabh-deep** [comment]
Hey @NickNaso!

Thank you for your response. We have tried `pkg` and `nexe`, as I mentioned originally in my post. While they work for the most part in creating the binary, but the binaries are invalid from the signing perspective, especially for the MacOS.

I will give `boxnode` a shot and see if that creates valid macOS binaries.

Thanks again!

**Comment by aduh95** [comment]
Also, you can check out [Electron](https://www.electronjs.org/), although that's probably a bit overkill for what you are trying to do, it should give you all the tools you need.
